packages/neon_framework/lib/src/blocs/user_status.dart

```diff
@@ -41,16 +74,33 @@ class _UserStatusBloc extends InteractiveBloc implements UserStatusBloc {
   @override
   void dispose() {
     timer.cancel();
+    unawaited(status.close());
     unawaited(statuses.close());
+    unawaited(predefinedStatuses.close());
     super.dispose();
   }
 
+  @override
+  final status = BehaviorSubject<Result<user_status.$PublicInterface>>();
+
   @override
   final statuses = BehaviorSubject<Map<String, Result<user_status.$PublicInterface>>>.seeded({});
 
+  @override
+  final predefinedStatuses = BehaviorSubject<Result<BuiltList<user_status.Predefined>>>();
+
   @override
   Future<void> refresh() async {
-    await Future.wait(statuses.value.keys.map((username) => load(username, force: true)));
+    await Future.wait([
+      RequestManager.instance.wrapNextcloud(
+        account.id,
+        'user_status-predefined-statuses',
+        predefinedStatuses,
+        account.client.userStatus.predefinedStatus.findAllRaw(),
+        (response) => response.body.ocs.data,
+      ),
+      ...statuses.value.keys.map((username) => load(username, force: true)),
```

Comment:
This is really dangerous.
The load function is asynchonous so the statuses subject can change between calling and execution.
Please just map over the entries and give the load function both the username and result (similar to `updateStatus`) or the MapEntry of them.

This cleans up a lot of the checks you needed to do there.

Reply:
I understand what you mean, but I'm not sure if I understand how you would like to see it work.